pkg/asset/agent/image/oc.go

```diff
@@ -169,6 +176,62 @@ func (r *release) getImageFromRelease(imageName string) (string, error) {
 
 	return image, nil
 }
+func (r *release) validateReleaseImageArch(architecture string) error {
+	// This requires the 'oc' command so make sure its available
+	_, err := exec.LookPath("oc")
+	var cmd string
+	if err != nil {
+		logrus.Error("Unable to validate release image architecture: \"oc\" command is not found")
+		return err
+	}
+
+	cmd = fmt.Sprintf(templateGetReleaseInfo, r.releaseImage)
+	image, err := execute(r.executer, r.pullSecret, cmd)
+	if err == nil {
+		type ReleaseConfig struct {
+			Architecture string `json:"architecture,omitempty"`
+		}
+		type ReleaseMetaDataMetaData struct {
+			ReleaseArch string `json:"release.openshift.io/architecture,omitempty"`
+		}
+		type ReleaseMetaData struct {
+			Metadata *ReleaseMetaDataMetaData `json:"metadata,omitempty"`
+		}
+		type ReleaseInfo struct {
+			Image    string           `json:"image"`
+			Metadata *ReleaseMetaData `json:"metadata,omitempty"`
+			Config   *ReleaseConfig   `json:"config,omitempty"`
+		}
+		var releaseimagedata ReleaseInfo
+		err := json.Unmarshal([]byte(image), &releaseimagedata)
+		if err != nil {
+			logrus.Errorf("can't read oc adm release info for %s ", r.releaseImage)
+			return err
+		}
+
+		// Check the release image to see if it is multi.
+		if releaseimagedata.Metadata.Metadata != nil {
+			if releaseimagedata.Metadata.Metadata.ReleaseArch == "multi" {
+				err = fmt.Errorf("multi architecture release image %s is not supported", r.releaseImage)
```

Comment:
This is just because we know we need to make more changes to make this work, I assume? A comment to that effect might be helpful for future-us.

Specifically, IIRC, we have to make changes to make it work when the cluster cpu arch is different from the host the installer is running on. Potentially we could limit the error to just that case.

Reply:
> This is just because we know we need to make more changes to make this work, I assume? A comment to that effect might be helpful for future-us.

Correct, we will tackle using the `multi` payload in 4.15. The changes are mostly adding '--filter-by-os' to the oc adm release command.

> Specifically, IIRC, we have to make changes to make it work when the cluster cpu arch is different from the host the installer is running on. Potentially we could limit the error to just that case.

This is true for BMIPI, but not agent. An x86 client can create a valid agent.aarch.iso as long as the release payload is arm64 (or in the future multi).

Reply:
right, I mean that's the case where `--filter-by-os` is needed. if the arch happens to match I think it will pull the right image even without `--filter-by-os`?

Reply:
correct, but then the story is you can only create an arm cluster with an x86 client IF you use a single arch arm payload. If using the multi payload the client machine has to match the architecture of the target nodes. At this time, we'd rather say multi isn't supported at install time, and then bring the multi payload to 'parity'.